File: tools/update.js
const fs = require('fs');

let content = fs.readFileSync('Lection3.html', 'utf8');

// 1. Clean up Tables 3.1 and 3.2 inline styles
// Remove all inline style="padding: 1pt 4pt; font-size: 10.5pt; height: 14pt;" from th and td
content = content.replace(/ style="padding: 1pt 4pt; font-size: 10\.5pt; height: 14pt;"/g, '');

// Remove inline zebra coloring (CSS handles it now)  
content = content.replace(/<tr style="background-color: #F2F7FC;">/g, '<tr>');

// 2. Add table-narrow class to Tables 3.1 and 3.2 wrappers
content = content.replace(
  /(<p class="table-caption"><strong>Таблиця 3\.1\.<\/strong>[^<]+<\/p>)\n<div class="table-responsive">/g,
  '$1\n<div class="table-responsive table-narrow">'
);
content = content.replace(
  /(<p class="table-caption"><strong>Таблиця 3\.2\.<\/strong>[^<]+<\/p>)\n<div class="table-responsive">/g,
  '$1\n<div class="table-responsive table-narrow">'
);

// 3. Fix Mermaid syntax: replace -- (double dash link) with --- (triple dash)
// Only within mermaid blocks. The pattern FOBx -- FBTx should be FOBx --- FBTx
content = content.replace(/FOB1 -- FBT1/g, 'FOB1 --- FBT1');
content = content.replace(/FOB2 -- PLC2/g, 'FOB2 --- PLC2');
content = content.replace(/FOB8 -- FBT8/g, 'FOB8 --- FBT8');

// 4. Remove old inline cellspacing/cellpadding/style from calculation tables
content = content.replace(/ cellspacing="0" cellpadding="1" style="border-collapse: collapse;"/g, '');

// 5. Fix the nav container to use 1400px instead of 960px
content = content.replace(/<div style="max-width: 960px;/g, '<div style="max-width: 1400px;');

fs.writeFileSync('Lection3.html', content, 'utf8');
console.log('Lection3.html updated successfully!');

// Also fix Lection1 and Lection2 nav containers if they have 960px
for (const file of ['Lection1.html', 'Lection2.html']) {
  let c = fs.readFileSync(file, 'utf8');
  c = c.replace(/<div style="max-width: 960px;/g, '<div style="max-width: 1400px;');
  fs.writeFileSync(file, c, 'utf8');
  console.log(file + ' nav updated.');
}

File: tools/update.py
import re
import os

def update_file(path, replacements):
    with open(path, 'r', encoding='utf-8') as f:
        content = f.read()
    
    for old, new in replacements:
        content = re.sub(old, new, content)
        
    with open(path, 'w', encoding='utf-8') as f:
        f.write(content)

# Lection 2 updates
l2_replacements = [
    (r'<p><em>План вивчення:</em><br>\s*1\.\s*(.+?)<br>\s*2\.\s*(.+?)<br>\s*3\.\s*(.+?)<br>\s*4\.\s*(.+?)</p>',
     r'<h3 class="center">План вивчення:</h3>\n<ol>\n  <li>\1</li>\n  <li>\2</li>\n  <li>\3</li>\n  <li>\4</li>\n</ol>'),
]
update_file('Lection2.html', l2_replacements)

# Lection 3 updates
l3_replacements = [
    # Plan
    (r'<p><em>План вивчення:</em><br>\s*1\.\s*(.+?)<br>\s*2\.\s*(.+?)<br>\s*3\.\s*(.+?)</p>',
     r'<h3 class="center">План вивчення:</h3>\n<ol>\n  <li>\1</li>\n  <li>\2</li>\n  <li>\3</li>\n</ol>'),
    
    # Table Responsive
    (r'(<p class="table-caption"><strong>Таблиця 3\.\d\.</strong>.+?</p>)\s*<table class="sim-table"',
     r'\1\n<div class="table-responsive">\n<table class="sim-table"'),
    (r'</table>', r'</table>\n</div>'),
    
    # Mermaid fix syntax
    (r'----\|', r'---|'),
    (r'---- ', r'--- '),
    
    # Tables Refactoring
    (r'<b style="color:#0d6efd"><span style="color:#0d6efd;font-size:10pt">● </span>', r'<b class="sym-in"><span class="sym-in">● </span>'),
    (r'<span style="color:#fd7e14;font-size:10pt">◆ </span>', r'<span class="sym-fbt">◆ </span>'),
    (r'<b style="color:#dc3545"><span style="color:#dc3545;font-size:10pt">● </span>', r'<b class="sym-out"><span class="sym-out">● </span>'),
    (r'<b style="color:#198754"><span style="color:#198754;font-size:10pt">● </span>', r'<b class="sym-tran"><span class="sym-tran">● </span>'),
    (r'<span style="color:#6f42c1;font-size:10pt">■ </span>', r'<span class="sym-plc">■ </span>'),
    (r'<b style="color:#dc3545; font-size:13pt"><span style="color:#dc3545;font-size:10pt">■ </span>', r'<b class="sym-out"><span class="sym-out">■ </span>'),
    (r'<span style="color:#0d6efd;font-size:10pt">● </span>', r'<span class="sym-in">● </span>'),
    (r'<span style="color:#dc3545;font-size:10pt">● </span>', r'<span class="sym-out">● </span>'),
    (r'<span style="color:#198754;font-size:10pt">● </span>', r'<span class="sym-tran">● </span>'),
    (r'<span style="color:#dc3545;font-size:10pt">■ </span>', r'<span class="sym-out">■ </span>'),
]
update_file('Lection3.html', l3_replacements)
print("Updated successfully!")
